Logstash/logstash-input-azuretopicthreadable/lib/logstash/inputs/azuretopicthreadable.rb

```diff
@@ -47,9 +58,12 @@ def process(output_queue, pid)
           codec.decode(message.body) do |event|
           decorate(event)
           output_queue << event
-        end # codec.decode
-        # delete the message after reading it
-        azure_service_bus.delete_subscription_message(message)
+          # delete the message after reading it
+          azure_service_bus.delete_subscription_message(message)
+        end
```

Comment:
@addp009, your do |event| is not properly closed. Try add an 'end' before 'else' statement, at the same time, remove the one after the end of else like this:
``` ruby
    while !stop?
      begin
        # check if we have a message in the subscription
        message = azure_service_bus.receive_subscription_message(@topic ,@subscription, { :peek_lock => true, :timeout => 1 } )
        if message
          # decoding returns a yield
          codec.decode(message.body) do |event|
            decorate(event)
            output_queue << event
            # delete the message after reading it
            azure_service_bus.delete_subscription_message(message)
          end # end of do
        else
          Stud.stoppable_sleep(@thread_sleep_time) { stop? } #topic is probably empty. sleep. 
        end
      rescue LogStash::ShutdownSignal => e
        raise e
      rescue => e
        @logger.error("Oh My, An error occurred. Thread id:" + pid.to_s, :exception => e)
        if message and message.delivery_count > @deliverycount
          azure_service_bus.delete_subscription_message(message)
        end
        Stud.stoppable_sleep(@thread_sleep_time) { stop? }
      end # end of rescues
    end # end of while
  end # def process
```